csrc/device_lower/lower2device.cpp

```diff
@@ -416,6 +416,8 @@ void GpuLower::lower(Fusion* fusion) {
   const auto exprs_sorted = reorderExprsForComputeAt();
   dumpExprsIfEnabled(exprs_sorted, "reorderExprsForComputeAt");
 
+  commonScalarMap().initialize(exprs_sorted);
```

Comment:
Does this move change anything?

Reply:
Just making sure index hoisting can be used for earlier passes such as loop nest generation, because `ForLoop:: simplifiedStop` now uses index hoisting.